bundles/pcmjava/tools.vitruv.applications.pcmjava.pojotransformations.java2pcm/src/tools/vitruv/applications/pcmjava/pojotransformations/java2pcm/Java2PcmClassifier.reactions

```diff
@@ -29,7 +28,7 @@ reaction PackageCreated {
 	with newValue.name === null || (!newValue.name.contains("contracts") && !newValue.name.contains("datatypes"))
 		
 	call {
-		createArchitecturalElement(newValue, getLastPackageName(newValue.name), getRootPackageName(newValue.name))
+		createOrFindArchitecturalElement(newValue, getLastPackageName(newValue.name))
 		createOrFindRepository(newValue, newValue.name, "package_root")
```

Comment:
It is hard to understand, why it is not possible that both an architectural element and a repository may get created after a packages was inserted (or the developer gets asked for creating both). Taking a deeper look into the implementation clarifies that both cases are mutually exclusive, but it's sort of hidden deeper in the implementation. I'll make a note that we should add some documentation here or maybe do some refactoring to clarify the behavior without further documentation.

Reply:
I agree, it is definitely something that is not obvious here.